File: 0242-valid-anagram/0242-valid-anagram.ts
function isAnagram(s: string, t: string): boolean {
     if(s.length != t.length) return false;
    const count = new Map();
    for(let i of s){
        count.set(i,(count.get(i)||0) + 1 );
    }
    for(let j of t){
        if(!count.has(j)) return false;
        count.set(j,count.get(j) - 1);
        if(count.get(j) < 0) return false;
    }
    return true;
};

File: 0242-valid-anagram/0242-valid-anagram.cs
public class Solution {
    public bool IsAnagram(string s, string t) {
        if(s.Length != t.Length) return false;
        Dictionary<char, int> count = new Dictionary<char,int>();
        foreach(char i in s){
            if(!count.ContainsKey(i)) count[i] = 0;
            count[i] ++;
        }
        foreach(char i in t){
            if(!count.ContainsKey(i)) return false;
            count[i]--;
            if(count[i] < 0) return false;
        }
        return true;
    }
}